integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -374,12 +374,15 @@ object CarbonDataRDDFactory {
               .getTableInfo
               .getFactTable
               .getListOfColumns
-              .asScala.filterNot(col => col.isInvisible || col.getColumnName.contains("."))
+              .asScala
+              .filterNot(col => col.isInvisible || col.isIndexColumn ||
+                                col.getColumnName.contains("."))
```

Comment:
If "col.getColumnName.contains(".")" is to check for complex columns then please use col.isComplexColumn instead.

Reply:
Modified